core/test/loader/atom.test.ts

```diff
@@ -457,3 +467,118 @@ test('parses media', () => {
     }
   )
 })
+
+test('detects pagination with rel="next" link', () => {
+  let $store = loaders.atom.getPosts(
+    createDownloadTask(),
+    'https://example.com/feed/',
+    exampleAtom(
+      `<?xml version="1.0"?>
+      <feed xmlns="http://www.w3.org/2005/Atom">
+        <link rel="self" href="https://example.com/feeds/posts.atom"/>
+        <link rel="next" href="https://example.com/feeds/posts-2024.atom"/>
+      </feed>`
+    )
+  )
+  equal($store.get().hasNext, true)
+})
+
+test('detects when there is no pagination', () => {
+  let $store = loaders.atom.getPosts(
+    createDownloadTask(),
+    'https://example.com/feed/',
+    exampleAtom(
+      `<?xml version="1.0"?>
+      <feed xmlns="http://www.w3.org/2005/Atom">
+      </feed>`
+    )
+  )
+  equal($store.get().hasNext, false)
+})
+
+test('loads first then second page', async () => {
+  let task = createDownloadTask()
+
+  let callCount = 0
+  let textSpy = spyOn(task, 'text', () => {
```

Comment:
We have our own HTTP request mocking system which can be used here.

You can find an example here: https://github.com/hplush/slowreader/blob/main/core/test/pages/add.test.ts

You need to replace `task.text()` to `expectRequest()` call before the check.

Reply:
Done and added a section on request mocking to readme.